exporter/elasticsearchexporter/README.md

```diff
@@ -83,32 +83,52 @@ The Elasticsearch exporter supports the common [`sending_queue` settings][export
-- `logs_dynamic_index` (optional):
-  takes resource or log record attribute named `elasticsearch.index.prefix` and `elasticsearch.index.suffix`
-  resulting dynamically prefixed / suffixed indexing based on `logs_index`. (priority: resource attribute > log record attribute)
+
+- `logs_dynamic_index` (optional): uses resource or log record attributes to dynamically construct index name. See `mode` for details.
   - `enabled`(default=false): Enable/Disable dynamic index for log records
-- `metrics_index`: The [index] or [data stream] name to publish metrics to. The default value is `metrics-generic-default`.
+  - `mode` (default=`prefix_suffix`): defines how dynamic index name is constructed.
+    - `data_stream` - uses resource, scope or log record attributes `data_stream.dataset` and `data_stream.namespace`
+      to dynamically construct index name in the form `logs-${data_stream.dataset}-${data_stream.namespace}`.
+      Log record attributes take precedence over scope attributes, which take precedence over resource attributes.
+    - `prefix_suffix` - uses resource or log record attributes `elasticsearch.index.prefix` and `elasticsearch.index.suffix`
+      to dynamically construct index name in the form `${elasticsearch.index.prefix}${logs_index}${elasticsearch.index.suffix}`. (priority: resource attribute > log record attribute)
+
+- `metrics_index` (optional): The [index] or [data stream] name to publish metrics to. The default value is `metrics-generic-default`.
   ⚠️ Note that metrics support is currently in development.
-- `metrics_dynamic_index` (optional):
-  takes resource attributes named `elasticsearch.index.prefix` and `elasticsearch.index.suffix`
-  resulting dynamically prefixed / suffixed indexing based on `metrics_index`.
+
+- `metrics_dynamic_index` (optional): uses resource, scope or data point attributes to dynamically construct index name. See `mode` for details.
   ⚠️ Note that metrics support is currently in development.
-  - `enabled`(default=false): Enable/Disable dynamic index for metrics
+  - `enabled`(default=true): Enable/disable dynamic index for metrics
+  - `mode` (default=`data_stream`): defines how dynamic index name is constructed.
+    - `data_stream` - uses resource, scope or data point attributes `data_stream.dataset` and `data_stream.namespace`
+      to dynamically construct index name in the form `metrics-${data_stream.dataset}-${data_stream.namespace}`.
+      Data point attributes take precedence over scope attributes, which take precedence over resource attributes.
+    - `prefix_suffix` - uses resource, scope or data point attributes `elasticsearch.index.prefix` and `elasticsearch.index.suffix`
+      to dynamically construct index name in the form `${elasticsearch.index.prefix}${metrics_index}${elasticsearch.index.suffix}`.
+      Data point attributes take precedence over scope attributes, which take precedence over resource attributes.
```

Comment:
metrics prefix_suffix precedence is different to logs and traces. Is this deliberate?

Reply:
I see that you've created https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33725 but the question remains: is it worth it to have this exception / inconsistency for metrics prefix_suffix mode? Otherwise, does it make sense to break all of them at once when a decision is made in #33725 ?

Reply:
Yes, I created https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33725 because I think the current setting for prefix_suffix should be changed.

When introducing metrics, my thinking was to introduce the `prefix_suffix` mode for metrics using the correct precedence, to prevent a breaking change at least in metrics behavior.

I do see value in having those consistent between signals though. If you think we should rather keep `prefix_suffix` mode precedence consistent with other signals, I can change that.